From daf20b58743fc141bfe8b2381ecd113b96a0cbd1 Mon Sep 17 00:00:00 2001
From: Jason Ekstrand <jason.ekstrand@intel.com>
Date: Thu, 3 Aug 2017 09:48:55 -0700
Subject: [PATCH 37/39] UPSTREAM: i965/miptree: Set supports_fast_clear = false
 in make_shareable

The make_shareable function deletes the aux buffer and then whacks
aux_usage to ISL_AUX_USAGE_NONE but not unsetting supports_fast_clear.
Since we only look at supports_fast_clear to decide whether or not to do
fast clears, this was causing assertion failures.

Reported-by: Tapani Plli <tapani.palli@intel.com>
Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=101925
Reviewed-by: Kenneth Graunke <kenneth@whitecape.org>
Reviewed-by: Tapani Plli <tapani.palli@intel.com>
(cherry picked from commit e7a52cc381d275b4ab8ee2fb230e32ab97090daf)

BUG=b:64614087
TEST=Desmos Graphing Calculator does not crash on startup

Change-Id: Iabeefa97ecde85ff955a85d95c6851eab529e924
Signed-off-by: Tomasz Figa <tfiga@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/630900
Tested-by: Dmytro Chystiakov <dmytro.chystiakov@intel.corp-partner.google.com>
Reviewed-by: Chad Versace <chadversary@chromium.org>
---
 src/mesa/drivers/dri/i965/intel_mipmap_tree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mesa/drivers/dri/i965/intel_mipmap_tree.c b/src/mesa/drivers/dri/i965/intel_mipmap_tree.c
index ed7cb8e..c1f18a7 100644
--- a/src/mesa/drivers/dri/i965/intel_mipmap_tree.c
+++ b/src/mesa/drivers/dri/i965/intel_mipmap_tree.c
@@ -2719,6 +2719,7 @@ intel_miptree_make_shareable(struct brw_context *brw,
    }
 
    mt->aux_usage = ISL_AUX_USAGE_NONE;
+   mt->supports_fast_clear = false;
 }
 
 
-- 
2.7.4

